services/controllerhost/replicaStatsMRUCache.go

```diff
@@ -127,82 +132,134 @@ func (c *replicaStatsMRUCache) get(extent extentID, store storeID) (val *shared.
 		if len(foundStore) != 0 {
 			lclLg.WithField(`substituteStore`, common.FmtStor(string(foundStore))).
 				Error(`Could not get stats for specified store; substituting`)
-			return c.get(extent, foundStore)
+			return psmap.m[foundStore] // Might be nil
 		}
 
 		lclLg.Error(`Could not get stats for specified store`)
 		return nil
-	} else if val != nil {
-		c.put(extent, store, val)
+	} else if extStats != nil {
+		val = makeCompactExtentStats(extStats)
+		// In the typical case, all store extents had planned sealing, meaning that their available sequences are the same
+		// Map the value pointer to the same compactExtentStats in this typical case, to reduce memory usage by nearly 66%
+	dedup:
+		for dedupStore, val2 = range psmap.m {
+			if store != dedupStore && val2 != nil && val2.seqAvailable == val.seqAvailable {
+				val = val2
+				break dedup
+			}
+		}
+
+		// psmap is a pointer into the cache, so we can simply modify it here
+		val.consumerGroupVisibility = psmap.singleCGVisibility
+		psmap.m[store] = val
 	}
 
 	return
-}
 
-func (c *replicaStatsMRUCache) putSingleCGVisibility(extent extentID, singleCGVisibility string) {
-	if len(extent) == 0 || len(singleCGVisibility) == 0 {
-		return
+ttlCheck:
+
+	// Open extents should only be valid for a single scan; All types of sealed extent should have indefinite retention
+	if val != nil && val.status == shared.ExtentStatus_OPEN && time.Since(val.cacheTime) > IntervalBtwnScans {
```

Comment:
if the TTL is intervalBtwnScans, is the purpose of this cache to avoid additional lookups when there are multiple consumer-groups for the same destination ?